thoth/storages/graph/postgres.py

```diff
@@ -3852,8 +3853,22 @@ def create_python_cve_record(
         package_version = self.normalize_python_package_version(package_version)
         with self._session_scope() as session, session.begin(subtransactions=True):
             cve, _ = CVE.get_or_create(
-                session, cve_id=record_id, version_range=version_range, advisory=advisory, cve_name=cve
+                session,
+                cve_id=record_id,
+                version_range=version_range,
+                advisory=advisory,
+                cve_name=cve,
+                # aggregated_at=datetime.now(),
             )
+            if cve.aggregated_at is None:
```

Comment:
@fridex WDYT about this approach? The idea is to get or create a CVE instance based on all parameters but `aggregated_at`. As we indicate `nullable=True`, the instance should have `None` if it is newly created. Once we detect a new row data, then we could set `aggregated_at` attribute of the object to `datetime.now()` and next you could just find that row based on `cve.id` which is primary key I guess, then update that record.

Reply:
The issue is in the `get_or_create` which does exact matching. Each time it will go through the records in the database and will find out there is no record that would have:

```python
                cve_id=record_id,
                version_range=version_range,
                advisory=advisory,
                cve_name=cve,
                aggregated_at=None
```

... so it will be created. The code below will set `aggregated_at` and insert such record. In the next round, there will be performed exact matching again (with `aggregated_at` set to `None`) so there will be no record created (again). You can try to replicate this if you run the CVE syncing logic multiple times.

Reply:
Should I really use `get_or_create`? What if I create a static method something like `get` in `BaseExtension` class which would check if the record exists on specific attribute values?

Reply:
You can directly querying logic as provided by sqlalchemy methods. [Here for example, we query package extract runs](https://github.com/thoth-station/storages/blob/0e0ee17a00f8aa0bc726e477344ee107b5e22090/thoth/storages/graph/postgres.py#L344-L352). Docs are available here - https://docs.sqlalchemy.org/en/13/orm/query.html